src/components/EditableToDo.vue

```diff
@@ -0,0 +1,235 @@
+        <p class="text-xs">{{ formatShortDate(date) }}</p>
+      </div>
+      <div class="self-start mt-4 mb-4">
+        <textarea
+          v-model="currentText"
+          placeholder="Lorem ipsum"
+          class="border-none h-[70.27px] w-[224.45px] lg:h-[103px] lg:w-[455.1px] text-start text-sm lg:text-lg text-grey"
+          :class="currentOpacity"
+          :disabled="isDropDownShown"
+        ></textarea>
+      </div>
+      <div class="text-sm flex flex-row font-semibold space-x-2">
+        <button
+          class="text-white w-[61px] h-[27px] rounded-lg bg-green"
+          :class="currentOpacity"
+          :disabled="isDropDownShown"
+          @click="saveToDo"
+        >
+          Save
+        </button>
+        <button
+          class="rounded-lg h-[27px] w-[70px] bg-buttonGray"
+          :class="currentOpacity"
+          :disabled="isDropDownShown"
+          @click="deleteToDo"
+        >
+          Delete
+        </button>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+  import { computed, ref, PropType } from 'vue';
+  import { Priority, ToDoItem } from '../models/todoitem-model';
+  import { formatShortDate } from '../utils/date-formatting';
+  import { getItemPriority } from '../utils/item-priority';
+  import ModalButton from './ModalButton.vue';
+
+  const props = defineProps({
+    item: Object as PropType<ToDoItem>,
+    index: {
+      type: Number,
+      default: -1,
+    },
+  });
+
+  const emit = defineEmits<{
+    (e: 'removeItem', ind: Number): void;
+    (e: 'addToDo', newToDo: ToDoItem): void;
+    (e: 'hideForm'): void;
+    (e: 'updateToDo', ind: Number, newToDo: ToDoItem): void;
+  }>();
+
+  const isDropDownShown = ref(false);
+
+  const currentOpacity = computed(() => {
+    if (isDropDownShown.value) {
+      return 'opacity-30' as const;
+    }
+    return '' as const;
+  });
+
+  const currentTitle = ref(props.item?.title || '');
+
+  const currentText = ref(props.item?.text || '');
+
+  const currentPriority = ref(props.item?.priority || Priority.low);
+
+  const priorityClass = computed(() => getItemPriority(currentPriority.value));
+
+  const date = new Date();
+
+  const priorityList = [Priority.low, Priority.medium, Priority.high];
+
+  const isPopUpShown = ref(false);
+
+  const positionOfDropDownButton = computed(() => {
+    if (isPopUpShown.value) {
+      return '' as const;
+    }
+    return 'relative';
+  });
+
+  const dropDownStyle = computed(() => {
+    if (isDropDownShown.value === true) {
+      return 'text-black bg-white border-2 border-black' as const;
+    }
+    return priorityClass.value;
+  });
+
+  function styleOfPriorityButton(ind: number) {
+    if (currentPriority.value === priorityList[ind]) {
+      return 'border border-black' as const;
+    }
+    return '';
+  }
+
+  function hidePopUp() {
+    isPopUpShown.value = false;
+  }
+
+  function clickAway() {
+    emit('hideForm');
+  }
+
+  function changeDropDownShown() {
```

Comment:
Please rename the function to changeDropDownVisibility or toggleDropDownVisibility